tasks.c

```diff
@@ -2854,15 +2854,18 @@ void vTaskStartScheduler( void )
          * starts to run. */
         portDISABLE_INTERRUPTS();
 
-        #if ( configUSE_NEWLIB_REENTRANT == 1 )
+        #if ( ( configUSE_NEWLIB_REENTRANT == 1 ) && ( configNEWLIB_REENTRANT_IS_DYNAMIC == 0 ) )
             {
                 /* Switch Newlib's _impure_ptr variable to point to the _reent
                  * structure specific to the task that will run first.
                  * See the third party link http://www.nadler.com/embedded/newlibAndFreeRTOS.html
-                 * for additional information. */
+                 * for additional information.
+                 *
+                 * Note: Updating the _impure_ptr is not required when Newlib is compiled with
+                 * __DYNAMIC_REENT__ enabled. The port should provide __getreent() instead. */
```

Comment:
Will the definition of `__getreent` function be same for all ports? 

```
struct _reent* __getreent(void) {
    // No lock needed because if this changes, we won't be running anymore.
    TCB_t *pxCurTask = xTaskGetCurrentTaskHandle();
    if (pxCurTask == NULL) {
        // No task running. Return global struct.
        return _GLOBAL_REENT;
    } else {
        // We have a task; return its reentrant struct.
        return &pxCurTask->xNewLib_reent;
    }
}
```

Is `_GLOBAL_REENT` from newlib?

Reply:
@aggarg

> Will the definition of __getreent function be same for all ports?

I can't say for sure.  One thing that could differ between ports is how `__getreent()` handles being called before `vTaskStartScheduler()` is called. In ESP-IDF, we need to support Newlib calls from the only CPU 0 before `vTaskStartScheduler()`. However, I'm not sure if other ports also want that behavior.

> Is _GLOBAL_REENT from newlib?

Yes, it is [provided in `reent.h`](https://github.com/bminor/newlib/blob/03e815a91b1a1f94ce0cfd6ff64f50ae40ed740c/newlib/libc/include/sys/reent.h#L863) and it points to the global reentrancy structure provided by Newlib.

Reply:
> In ESP-IDF, we need to support Newlib calls from the only CPU 0 before vTaskStartScheduler().

And I believe that is the reason you are able to return `_GLOBAL_REENT`. Otherwise, you will need a reent per core, right?

Lets keep it in port only then.